finagle-redis/src/main/scala/com/twitter/finagle/redis/ClusterCommands.scala

```diff
@@ -0,0 +1,230 @@
+package com.twitter.finagle.redis
+
+import com.twitter.finagle.redis.protocol.StatusReply
+import com.twitter.io.{Buf, ByteReader}
+import com.twitter.util.Future
+import com.twitter.finagle.redis.protocol._
+import com.twitter.finagle.redis.util.{BufToString, ReplyFormat, StringToBuf}
+import java.net.InetSocketAddress
+import java.nio.charset.StandardCharsets.UTF_8
+
+private[redis] trait ClusterCommands { self: BaseClient with BasicServerCommands =>
+
+  private def toInfoMap(buf: Buf): Map[String, String] = {
+    def readAll(acc: Map[String, String], reader: ByteReader): Map[String, String] = {
+      reader.remainingUntil('\r') match {
+        case -1 => acc
+        case n =>
+          val newAcc = reader.readString(n, UTF_8).split(":") match {
+            case Array(key, value) => acc + (key -> value)
+            case _ => acc
+          }
+
+          // jump past the '\r\n'
+          reader.skip(2)
```

Comment:
How do we know `\n` follows `\r`? I don't see a check for that.

Reply:
Its an assumption from the protocol: https://redis.io/topics/protocol#bulk-string-reply